ext/openssl/ossl_bn.c

```diff
@@ -960,6 +960,29 @@ ossl_bn_uminus(VALUE self)
     return obj;
 }
 
+/*
+ * call-seq:
+ *   bn.abs -> aBN
+ */
+static VALUE
+ossl_bn_abs(VALUE self)
+{
+    VALUE obj;
+    BIGNUM *bn1, *bn2;
+
+    GetBN(self, bn1);
+    if (BN_is_negative(bn1)) {
+        obj = NewBN(cBN);
+        bn2 = BN_dup(bn1);
+        if (!bn2) { ossl_raise(eBNError, "BN_dup"); }
+        SetBN(obj, bn2);
+        BN_set_negative(bn2, 0);
+        return obj;
+    } else {
+        return self;
+    }
```

Comment:
`OpenSSL::BN` is a mutable object (`#set_bit!` exists, for example), so it should create a new instance of `OpenSSL::BN` even if the value is already positive.

Reply:
I was trying to track the behavior of unary plus.  I figured they should be the same, unary plus returns the same object, unary minus is always a copy.  abs did whatever was appropriate.  I know you're trying to imply that there are a ton of subtle bugs that come from "mutable primitives" which a BN still feels pretty Int like when you're using it for math....

Reply:
Good catch, the behavior of `OpenSSL::BN#+@` is a bug and needs to be fixed.

Reply:
I'll revise the PR to correct both to returning dups of their input